tests/core/featurizers/test_tracker_featurizer.py

```diff
@@ -49,48 +64,2238 @@ def test_convert_labels_to_ids(domain: Domain):
+
+    assert actual_features is not None
+    assert len(actual_features) == len(expected_features)
+
+    for actual, expected in zip(actual_features, expected_features):
+        assert compare_featurized_states(actual, expected)
+
+    expected_labels = np.array([[0, 15, 0, 12, 13, 0, 14]])
+    assert actual_labels is not None
+    assert len(actual_labels) == 1
+    for actual, expected in zip(actual_labels, expected_labels):
+        assert np.all(actual == expected)
+
+    # moodbot doesn't contain e2e entities
+    assert not any([any(turn_tags) for turn_tags in entity_tags])
+
+
+def test_featurize_trackers_ignore_action_unlikely_intent_with_full_dialogue_tracker_featurizer(
+    moodbot_domain: Domain, moodbot_features: Dict[Text, Dict[Text, Features]],
+):
+
+    tracker = DialogueStateTracker.from_events(
+        "default",
+        [
+            ActionExecuted(ACTION_LISTEN_NAME),
+            user_uttered("greet"),
+            ActionExecuted(ACTION_UNLIKELY_INTENT_NAME),
+            ActionExecuted("utter_greet"),
+            ActionExecuted(ACTION_LISTEN_NAME),
+            user_uttered("mood_unhappy"),
+            ActionExecuted(ACTION_UNLIKELY_INTENT_NAME),
+            ActionExecuted("utter_cheer_up"),
+            ActionExecuted("utter_did_that_help"),
+            ActionExecuted(ACTION_LISTEN_NAME),
+            user_uttered("deny"),
+            ActionExecuted(ACTION_UNLIKELY_INTENT_NAME),
+            ActionExecuted("utter_goodbye"),
+        ],
+        domain=moodbot_domain,
     )
-    state_features, labels, entity_tags = tracker_featurizer.featurize_trackers(
-        [tracker], moodbot_domain, RegexInterpreter()
+    state_featurizer = SingleStateFeaturizer()
+    tracker_featurizer = FullDialogueTrackerFeaturizer(state_featurizer)
+
+    actual_features, actual_labels, entity_tags = tracker_featurizer.featurize_trackers(
+        [tracker],
+        moodbot_domain,
+        RegexInterpreter(),
+        ignore_action_unlikely_intent=True,
     )
 
-    assert state_features is not None
-    assert len(state_features) == 1
-    assert labels is not None
-    assert len(labels) == 1
+    expected_features = [
+        [
+            {},
+            {
+                INTENT: [moodbot_features["intents"]["greet"]],
+                ACTION_NAME: [moodbot_features["actions"][ACTION_LISTEN_NAME]],
+            },
+            {ACTION_NAME: [moodbot_features["actions"]["utter_greet"]]},
+            {
+                INTENT: [moodbot_features["intents"]["mood_unhappy"]],
+                ACTION_NAME: [moodbot_features["actions"][ACTION_LISTEN_NAME]],
+            },
+            {ACTION_NAME: [moodbot_features["actions"]["utter_cheer_up"]]},
+            {ACTION_NAME: [moodbot_features["actions"]["utter_did_that_help"]]},
+            {
+                INTENT: [moodbot_features["intents"]["deny"]],
+                ACTION_NAME: [moodbot_features["actions"][ACTION_LISTEN_NAME]],
+            },
+        ]
+    ]
+
```

Comment:
I just ordered them arbitrarily but consistently because the state is a dictionary. It won't always be the case that the intent follows the action. For instance,

```yaml
- ActionExecuted("action_listen")
- UserUttered("greet")
- ActionExecuted("utter_greet")
- ActionExecuted("utter_how_are_you")
```

would produce the following states (putting the state keys in chronological order) :

```json
[
    {}, 
    {
        "prev_action": {"action_name": "action_listen"},
        "user": {"intent": "greet"}       
    },
    {
        "user": {"intent": "greet"},
        "prev_action": {"action_name": "utter_greet"}
    },
    {
        "user": {"intent": "greet"},
        "prev_action": {"action_name": "utter_how_are_you"}
    },
]
```

I could reorder them this way if you think that would help. I don't think there is any Rasa convention around this.

Reply:
Ah, this is much more readable, Chris. As someone who is trying to understand what the code does, I think this order will help me a lot.

Reply:
But it's also a massive hassle to go back and change this, so I'll let you decide if it's worth it.

Reply:
I was able to make this change pretty quickly.